cmd/kubectl-gadget/seccompadvisor.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/api/v1alpha1"
+)
+
+var seccompAdvisorCmd = &cobra.Command{
+	Use:   "seccomp-advisor",
+	Short: "Generate seccomp policies based on recorded syscalls activity",
+}
+
+var seccompAdvisorMonitorCmd = &cobra.Command{
+	Use:   "start",
+	Short: "Start to monitor the system calls",
+	Run:   runSeccompAdvisorStart,
+}
+
+var seccompAdvisorReportCmd = &cobra.Command{
+	Use:   "stop",
+	Short: "Stop monitoring and report the policies",
+	Run:   runSeccompAdvisorStop,
+}
+
+var seccompAdvisorListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List existing seccomp traces",
+	Run:   runSeccompAdvisorList,
+}
+
+var (
+	seccompProfile string
+)
+
+func init() {
+	// Add generic information.
+	rootCmd.AddCommand(seccompAdvisorCmd)
+	utils.AddCommonFlags(seccompAdvisorCmd, &params)
+
+	seccompAdvisorCmd.AddCommand(seccompAdvisorMonitorCmd)
+	seccompAdvisorMonitorCmd.PersistentFlags().StringVarP(&seccompProfile, "seccomp-profile", "", "", "ExternalResource name")
+
+	seccompAdvisorCmd.AddCommand(seccompAdvisorReportCmd)
+	seccompAdvisorCmd.AddCommand(seccompAdvisorListCmd)
+}
+
+// runSeccompAdvisorStart starts monitoring of syscalls for the given
+// parameters.
+func runSeccompAdvisorStart(cmd *cobra.Command, args []string) {
+	if params.Podname == "" {
```

Comment:
We don't need to specify the podname if the `outputMode` is `ExternalResource`. 

In general, I don't think we should do this kind of check in the CLI side but we should do it only on the container side to avoid duplicating stuff. If the filter is wrong, it will be notified by the container side through the `OperationError` field of the trace.